### PR TITLE
update dependencies and prettierrc

### PR DESCRIPTION
**Pull request by tsiegleauq**
Updates out client components and adjusts the `prettierrc` for new HTML indentation (HTML is still new for prettier)

disable `--aot` for `ng serve`  and enable it for `ng build` which results in significant faster (incremental) compilation time during development.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,32 +3,33 @@
     "version": "0.0.0",
     "scripts": {
         "ng": "ng",
-        "start": "ng serve --proxy-config proxy.conf.json --host=0.0.0.0 --aot",
-        "build": "ng build",
+        "start": "ng serve --proxy-config proxy.conf.json --host=0.0.0.0",
+        "build": "ng build --prod --aot",
```
Comment (jsaalfeld) [inline client/package.json]: shouldn't we have a build for _normal_ build and one for _prod_?

Comment (tsiegleauq) [reply client/package.json]: you can use ng build if you have to. this build command was run using `npm run build`

**Comment by tsiegleauq** [comment]
+ updated key events since they raised a deprecation warning and removed some unused imports

**Comment by jsaalfeld** [comment]
If anyone want's to dive into the rabbit whole of the deprication of the `no-unused-variable`-thing in tslint: https://github.com/palantir/tslint/issues/4100

**Comment by tsiegleauq** [comment]
updated syntax for `load children`. This fixed the AOT "cannot find the module" error.

**Comment by jsaalfeld** [comment]
Is the `load children` syntax now the way to go for the router in general? If so, then we should change everything to the `load children` syntax.

**Comment by tsiegleauq** [comment]
I suppose it is, in case of lazy loaded children. I thought I changed them all. Did I forget some?

**Comment by jsaalfeld** [comment]
didn't look too deep in yet, will check tomorrow morning before you visit the office. Either it will be merged by then or I'll add some requests.

**Comment by jsaalfeld** [comment]
This looks like the error described in this issue: https://github.com/angular/angular/issues/27023

(I don't know if the workaround viable somehow?)

**Comment by FinnStutzenstein** [comment]
So this is a webpack issue... https://github.com/angular/angular-cli/issues/9680#issuecomment-413158834

**Comment by tsiegleauq** [comment]
Found this interesting issue:
https://github.com/jasmine/jasmine/issues/1523

**Pull request merged**